.github/workflows/lint.yml

```diff
@@ -12,13 +12,13 @@ jobs:
     steps:
       - uses: actions/setup-go@v4
         with:
-          go-version: "1.20"
+          go-version: "1.21"
```

Comment:
do we need the upgrade? if yes, please ensure this consistent across

Reply:
linter fails in my local. I didn't update the go version for the implementation, to avoid risk of unexpected error (though small risk as it is Go 1.20 to 1.21). will update all into `1.21` unless there's further feedback.

Reply:
updated